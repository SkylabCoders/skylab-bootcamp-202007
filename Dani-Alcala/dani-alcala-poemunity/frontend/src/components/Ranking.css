.ranking {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.ranking__title {
  color: #4F5D73;
}

.ranking__subtitle {
  color: #4F5D73;
}

.ranking__picture-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.ranking__picture-wrap {
  position: relative;
  height: 60px;
  width: 92px;
}

.ranking__picture-wrap:hover .ranking__picture-description {
  visibility: visible;
  opacity: 1;
  cursor: default;
}

.ranking__picture-wrap:hover .ranking__picture {
  visibility: hidden;
  opacity: 1;
  cursor: default;
}

.ranking__picture {
  border-radius: 50%;
  height: 60px;
  width: 60px;
}

.ranking__picture-description {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  color: #000;
  visibility: hidden;
  opacity: 0;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-transition: opacity .2s, visibility .2s;
  transition: opacity .2s, visibility .2s;
}
/*# sourceMappingURL=Ranking.css.map */