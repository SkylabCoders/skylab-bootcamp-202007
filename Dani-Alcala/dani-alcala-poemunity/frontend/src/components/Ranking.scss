.ranking {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.ranking__title {
    color: #4F5D73;
}

.ranking__subtitle {
    color: #4F5D73;
}

.ranking__picture-container {
    display: flex;
}

.ranking__picture-wrap {
    position: relative;
    height: 60px;
    width: 92px;
}

.ranking__picture-wrap:hover .ranking__picture-description {
    visibility: visible;
    opacity: 1;
    cursor: default;
}

.ranking__picture-wrap:hover .ranking__picture {
    visibility: hidden;
    opacity: 1;
    cursor: default;
}

.ranking__picture {
    border-radius: 50%;
    height: 60px;
    width: 60px;
}

.ranking__picture-description {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    color: #000;
    visibility: hidden;
    opacity: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: opacity .2s, visibility .2s;
}