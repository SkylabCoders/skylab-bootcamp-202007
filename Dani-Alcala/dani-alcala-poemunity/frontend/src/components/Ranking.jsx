import React, {useState, useEffect} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import "./Ranking.scss";
import poemStore from "../stores/poemStore";
import { loadPoems } from "../actions/poemActions";

const useStyles = makeStyles({
  table: {
    minWidth: 50,
  },
});

export default function Ranking() {
  const classes = useStyles();
  const poemPoints = 3;
  const likePoints = 1;
  
  const [poems, setPoems] = useState(poemStore.getPoems());
  const [rank, setRank] = useState(poemStore.getRanking(poems, poemPoints, likePoints))

  useEffect(() => {
    poemStore.addChangeListener(onChange);
    if (poems.length === 0) loadPoems();
    else {
      onChange();
    }
    return () => poemStore.removeChangeListener(onChange);
  }, [poems.length], rank);

  function onChange() {
    setPoems(poemStore.getPoems());
    setRank(poemStore.getRanking(poems, poemPoints, likePoints));
  }

  return (
    <>
    
    <main className="ranking">
    <h3 className="ranking__title">Ranking de poetas</h3>
    <h5 className="ranking__subtitle">3 puntos por poema, 1 por like</h5>
    <TableContainer className="ranking__body" component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell align="center">Poetas</TableCell>
            <TableCell align="center">Puntos</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
        {Object.keys(rank).map((element, index) => (
            <TableRow key={index}>
              <TableCell className="ranking__picture-container" align="center" component="th" scope="row">
                {<div className="ranking__picture-wrap">
                  <img className="ranking__picture" src={rank[element].picture}/>
                  <p className="ranking__picture-description">{rank[element].author}</p>
                </div>}  
              </TableCell>
              <TableCell className="ranking__number" align="center">{rank[element].points}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
    </main>
  </>
  );
}
