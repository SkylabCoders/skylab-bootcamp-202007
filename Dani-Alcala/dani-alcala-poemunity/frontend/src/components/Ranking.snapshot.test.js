import React from 'react';
import renderer from 'react-test-renderer';
import Ranking from './Ranking';
import { BrowserRouter } from 'react-router-dom';

function renderRanking(arg) {
	const defaultProps = {
		match: {
			params: {}
		}
	};

	const props = { ...defaultProps, ...arg };
	return renderer.create(
		<BrowserRouter>
			<Ranking {...props} />
		</BrowserRouter>
	);
}

describe('Ranking', () => {
	let RankingTree;

	beforeEach(async () => {
		RankingTree = renderRanking();
	});

	it('should match without id', async () => {
		expect(RankingTree).toMatchSnapshot();
    });
})