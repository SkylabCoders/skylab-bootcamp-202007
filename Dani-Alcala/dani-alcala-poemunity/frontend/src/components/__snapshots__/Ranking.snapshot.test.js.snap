// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ranking should match without id 1`] = `
<main
  className="ranking"
>
  <h3
    className="ranking__title"
  >
    Ranking de poetas
  </h3>
  <h5
    className="ranking__subtitle"
  >
    3 puntos por poema, 1 por like
  </h5>
  <div
    className="MuiPaper-root MuiTableContainer-root ranking__body MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Poetas
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            Puntos
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</main>
`;
